core/domain/acl_decorators.py

```diff
@@ -461,21 +455,18 @@ def test_can_manage_collection_publish_status(
             raise base.UserFacingExceptions.PageNotFoundException
 
         if collection_rights.is_published():
-            if role_services.ACTION_UNPUBLISH_PUBLIC_COLLECTION in self.actions:
+            if (role_services.ACTION_UNPUBLISH_ANY_PUBLIC_ACTIVITY in
```

Comment:
Just wondering: why do we check all this here and not in check_can_publish_activity() below?

Edit: Ah, this is for unpublishing. I wonder, should we be doing something separate for "publish" vs "unpublish"? I can see different groups having different rights for that, similar to explorations. It also seems a bit messy/error-prone to deal with both here.